incubator/hnc/hack/crd_patches/singleton-enum-patch.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# Patch singleton metadata validation to each version, e.g.
+# 1) hierarchyconfiguration singleton:
+#            metadata:
+#              properties:
+#                name:
+#                  type: string
+#                  enum:
+#                    - hierarchy
+#              type: object
+# 2) hncconfig singleton:
+#            metadata:
+#              properties:
+#                name:
+#                  type: string
+#                  enum:
+#                    - config
+#              type: object
+sed -i -e "/ metadata:/ r hack/crd_patches/hierarchy_singleton_enum.yaml" config/crd/bases/hnc.x-k8s.io_hierarchyconfigurations.yaml
+sed -i -e "/ metadata:/ r hack/crd_patches/config_singleton_enum.yaml" config/crd/bases/hnc.x-k8s.io_hncconfigurations.yaml
```

Comment:
For those of us who don't know `sed` well 🙋 can you comment exactly what this does? I take it we assume there's only one line called `' metadata:'` in each file? If so, that seems like a reasonable assumption to document here as well.

Reply:
There are 3 lines. Commented why now, PTAL.

Reply:
Glad I asked :) Thanks!